xbmc/GUIInfoManager.cpp

```diff
@@ -4396,9 +4432,25 @@ void CGUIInfoManager::UpdateFPS()
   }
 }
 
+bool CGUIInfoManager::PlayingChannelMatchesDisplayedChannel() const
+{
+  bool bRet(g_application.m_pPlayer->IsPlaying());
+  if (bRet)
+  {
+    const CPVRChannelPtr currentlyDisplayedChannel(m_currentFile->GetPVRChannelInfoTag());
+    if (currentlyDisplayedChannel)
+    {
+      const CPVRChannelPtr currentlyPlayingChannel(g_PVRManager.GetCurrentChannel());
+      if (currentlyPlayingChannel && (*currentlyDisplayedChannel != *currentlyPlayingChannel))
+        bRet = false;
+    }
+  }
+  return bRet;
+}
```

Comment:
bool CGUIInfoManager::PlayingChannelMatchesDisplayedChannel() const
{
  return g_application.m_pPlayer->IsPlaying() &&
         m_currentFile->HasPVRChannel() &&
         *m_currentFile->GetPVRChannelInfoTag() == *g_PVRManager.GetCurrentChannel(); // there will always be a channel in pvrmanager while m_currentFile has a channel
}